mlir/test/Dialect/Linalg/fusion-elementwise-ops.mlir

```diff
@@ -977,3 +977,34 @@ module {
 //   CHECK-DAG:     %[[T3:.+]] = arith.addf %[[T2]], %[[B1]]
 //       CHECK:     linalg.yield %[[T3]] : f32
 //       CHECK:   return %[[GENERIC]]
+
+// -----
+
+#map = affine_map<(d0, d1, d2) -> (d1, d2)>
+#map1 = affine_map<(d0, d1, d2) -> (d0, d1, d2)>
+#map2 = affine_map<(d0, d1) -> (d0, d1 floordiv 4, d1 mod 4)>
+#map3 = affine_map<(d0, d1) -> (d0, d1)>
+
+// CHECK-DAG: [[$MAP0:#[a-zA-Z0-9_]*]] = affine_map<(d0, d1) -> (d0, d1)>
+// CHECK-DAG: [[$MAP1:#[a-zA-Z0-9_]*]] = affine_map<(d0) -> (d0 floordiv 4)>
+
+func.func @fuse_and_collapse(%arg0: tensor<3x4xi32>) -> tensor<2x12xi32> {
+  %1 = tensor.empty() : tensor<2x3x4xi32>
+  // CHECK: linalg.generic {
+  // CHECK: %[[INDEX1:[a-zA-Z0-9_]+]] = linalg.index 1 : index
+  // CHECK-NEXT: %[[MAP:[a-zA-Z0-9_]+]] = affine.apply #map1(%[[INDEX1]])
+  // CHECK-NEXT: %[[CAST:[a-zA-Z0-9_]+]] = arith.index_cast %[[MAP]] : index to i32
```

Comment:
Is the index_cast necessary to show the original bug, or could this test case also reproduce it without it (and using tensors with `index` element type instead)?